addons/account_facturx/models/account_move.py

```diff
@@ -239,6 +239,24 @@ def find_partner(partner_type):
                                 invoice_line_form.price_unit = float(line_elements[0].text) / float(quantity_elements[0].text)
                             else:
                                 invoice_line_form.price_unit = float(line_elements[0].text)
+                            # For Gross price, we need to check if a discount must be taken into account
+                            line_elements = element.xpath('.//ram:AppliedTradeAllowanceCharge',
+                                                          namespaces=tree.nsmap)
+                            if line_elements:
+                                line_elements = element.xpath(
+                                    './/ram:AppliedTradeAllowanceCharge/ram:CalculationPercent', namespaces=tree.nsmap)
+                                if line_elements:
+                                    invoice_line_form.discount = float(line_elements[0].text)
+                                else:
+                                    # if discount not available, it will be computed from the gross and net prices.
+                                    line_elements = element.xpath('.//ram:NetPriceProductTradePrice/ram:ChargeAmount',
+                                                                  namespaces=tree.nsmap)
+                                    if line_elements:
+                                        quantity_elements = element.xpath(
+                                            './/ram:NetPriceProductTradePrice/ram:BasisQuantity', namespaces=tree.nsmap)
+                                        gross_unit_price = float(line_elements[0].text) / float(quantity_elements[0].text) \
+                                            if quantity_elements else float(line_elements[0].text)
+                                        invoice_line_form.discount = (invoice_line_form.price_unit - gross_unit_price) / invoice_line_form.price_unit * 100.0
```

Comment:
What you're getting here is the net unit price :)

```suggestion
                                        net_unit_price = float(line_elements[0].text) / float(quantity_elements[0].text) \
                                            if quantity_elements else float(line_elements[0].text)
                                        invoice_line_form.discount = (invoice_line_form.price_unit - net_unit_price) / invoice_line_form.price_unit * 100.0
```

Reply:
Good catch ^^